src/oom/oomd-manager.c

```diff
@@ -349,6 +353,12 @@ static int monitor_swap_contexts_handler(sd_event_source *s, uint64_t usec, void
 
         assert(s);
 
+        /* Disable source until there are entries to monitor. */
+        if (hashmap_isempty(m->monitored_swap_cgroup_contexts)) {
```

Comment:
Maybe this should be an assertion? IIUC the timer is never enabled now unless there are swap entries to monitor. Hence, if the handler runs in that situation, it's a bug and we should not cover it up.

Reply:
+1 for assertion.
But the event source must be created disabled in `monitor_swap_contexts()`.

Reply:
Switched to `assert()` and ensured we start with `SD_EVENT_OFF`.

Reply:
I'm sorry, I didn't realize that a failure to switch the enabled state in `process_managed_oom_message()` is ignored. So this cannot be an assertion after all...

Reply:
This is still not addressed. The assertion will be triggered if we fail to disable the event source and the hashmap is empty.

Reply:
Based on the further discussion down below, please propagate the error in `sd_event_source_set_enabled`.

Reply:
Updated to propagate errno while logging error.